### ISSUE TITLE
document the code of the poroject to make it more readable

### ISSUE THREAD

**Issue by Krarilotus** [Sh0wdown/UnofficialCrusaderPatch#564]
We need documentation for our code.
I propose following the google styleguide:
https://google.github.io/styleguide/cppguide.html

specially just go to how to comment functions. Most functions are NOT SELFEXPLANATORY. And i spent now hours over hours to find out what just single functions do. Let alone a set of functions, called in a specific order. This must stop.

The bare minimum for me would be: to add a comment, that defines what inputs the function can take (YES also those from inherited classes, why would you inherit multiple classes and then loose track of possibilities for your functions by searching code around because you cant find out what each subclass now has?)  and also some basic explanation of what the functions is supposed to do.

Obviously you don't write down, what the function does, which is already immediately obvious in the code. You more so write down what the meaning of the function is.
So for example for a function that says: 

```
public void Activate(ChangeArgs args)
{
  this.labels.Clear();
  if (!editList.TrueForAll(e => e.Initialize(args)))
    return;
  editList.ForEach(e => e.Activate(args));
}
```
you would not need to explain that it looks for elements that are initialized and then activates them, you would instead explain waht activation means. What is an activated element and in what case would i call this function?

Ok, so i will add some comments after this release version, as i don't find time to now go through the code base and add a bunch of coments everywhere, but i will probably tidy up the code base a bit after the next release.

Hope to see more people following the basic guidelines of multiperson projects :)
lets do this!

**Comment by ByBurton** [comment]
Hahaha, I just think that's sooo ironic.
At work we have projects with thousands and thousands of lines of code, but only about a hundred lines of comments.

**Comment by GRhin** [comment]
How is that ironic? Or does krarilotus work with you?

**Comment by PodeCaradox** [comment]
We need to discuss this for the backend changes with the new Gui. So there will be no methods like this in future. More simple stuff you understand easily.

**Comment by Krarilotus** [comment]
We are a developer team with no guidelines really right now, which is just not good for a project of this scale, which gets bigger and bigger. Having no guidelines or documentation would be detrimental for the future development of the patch.

Good code speaks for itself, but for good code, you need to have a common ground of guidelines, which i would just default to the google stileguide. The google stile guide covers alot of ground and in case of questions of how to write code or comments, its an easy thing to look it up.

@ByBurton maybe your work should then take some pages out of the book of our development, or maybe even with that amount of code, your projects have other forms of organization to keep track of everything. Either way, we need in and out of project doumentation.
I am happy to discuss constructive opinions on this topic and take all the developers comments into consideration to form a common ground.

**Comment by PodeCaradox** [comment]
@Krarilotus 
For Clean code we should use the Code convention for C#:
https://docs.microsoft.com/de-de/dotnet/csharp/programming-guide/inside-a-program/coding-conventions
https://www.dofactory.com/reference/csharp-coding-standards

For Documentation of the Code we can use Comments that will get parse to xml, and with the xml we can create Document Files:
https://visualstudiomagazine.com/articles/2011/07/21/wcovb_xml-comments-part-1.aspx

**Comment by Krarilotus** [comment]
Yes thats perfect, exactly like that! I vote for this @PodeCaradox
